### PR TITLE
Added convert-string

### PR DESCRIPTION
**Pull request by hmajid2301**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/react-native-dotenv/tsconfig.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
         "module": "commonjs",
         "lib": ["es6"],
         "esModuleInterop": true,
-        "noImplicitAny": true,
+        "noImplicitAny": false,
```
Comment (weswigham) [inline types/react-native-dotenv/tsconfig.json]: Does this need to be changed? I'm not seeing how this change necessitates disabling `noImplicitAny`.

Comment (hmajid2301) [reply types/react-native-dotenv/tsconfig.json]: You are correct, this doesn't need to be changed :).

Comment (weswigham) [inline types/react-native-dotenv/tsconfig.json]: This still seems to be changed?

Comment (hmajid2301) [reply types/react-native-dotenv/tsconfig.json]: Umm ok weird, here it should be fixed now.

Comment (weswigham) [inline types/react-native-dotenv/tsconfig.json]: You keep pushing commits flipping the `convert-string` tsconfig, but it's the `react-native-dotenv` tsconfig I'm commenting on.

**Review by weswigham — CHANGES_REQUESTED** [review]

File: types/convert-string/tsconfig.json [hunk 2]
```diff
@@ -2,8 +2,8 @@
     "compilerOptions": {
         "module": "commonjs",
         "lib": ["es6"],
-        "noImplicitAny": true,
         "noImplicitThis": true,
+        "noImplicitAny": true,
```
Comment (weswigham) [inline types/convert-string/tsconfig.json]: OK, now you're just moving around the flag within the same file (and still not swapping back the flag in the other tsconfig) - why?

**Review by weswigham — CHANGES_REQUESTED** [review]

File: types/convert-string/tsconfig.json (hunk 2 above)
Comment (hmajid2301) [reply types/convert-string/tsconfig.json]: Apologies I've got really confused, didn't realised somehow this branch got mixed with `react-native-dotenv`. Hopefully this is fixed :) now.

**Review by weswigham — APPROVED** [review]

**Pull request merged**